sunxspex/sun_xspec.py

```diff
@@ -0,0 +1,349 @@
+              (not photons/cm$^2$/s/keV) i.e. it is the model spectrum integrated over the energy bin.
+        """
+
+        photon_energies=np.array(photon_energies)
+        internal_flux=np.zeros(photon_energies.shape)
+
+        # Constants
+        #const=constants.Constants()
+        mc2 = const.get_constant('mc2')
+        clight = const.get_constant('clight')
+        au = const.get_constant('au')
+        r0 = const.get_constant('r0')
+
+        # Max number of points
+        maxfcn = 2048
+
+        # Average atomic number
+        z = 1.2
+
+        # Relative error
+        rerr = 1e-4
+
+        # Numerical coefficient for photo flux
+        fcoeff = ((clight ** 2 / mc2 ** 4) / (4 * np.pi * au ** 2))
+        decoeff = 4.0 * np.pi * (r0 ** 2) * clight
+
+        # Create arrays for the photon flux and error flags.
+        #flux = np.zeros_like(photon_energies, dtype=np.float64)
+        iergq = np.zeros_like(photon_energies, dtype=np.float64)
+        try:
+            a0,p, eebrk, q, eelow, eehigh,_=params #why is this not consistent within xspec?
+        except ValueError:
+            a0,p, eebrk, q, eelow, eehigh=params
+
+        if eelow >= eehigh:
+            return list(photon_energies)[1:]
+
+        i, = np.where(np.logical_and(photon_energies < eehigh, photon_energies > 0)) #somewhat faster than &
+
+        if i.size > 0:
+            #logging.info(f"len(i)={i.size}, i[-1]={i[-1]}")
+            try:
+                internal_flux[i], iergq[i] = split_and_integrate0(model='thick-target', photon_energies=photon_energies[i],maxfcn=maxfcn, rerr=rerr, eelow=eelow,eebrk=eebrk, eehigh=eehigh, p=p, q=q, z=z,efd=False)
+            except ValueError:
+                flux[:]=[f*1e-10 for f in flux] #sometimes making it smaller might be better... but returning the original flux can also be misleading...have to figure out what to do about this
+                return flux
+
+            internal_flux = (fcoeff / decoeff) * internal_flux
+            #logging.info(f"{dt.now()} PARAMS {params}")
+            internal_flux[i]=internal_flux[i]*photon_energies[i]*a0*1e35
+            #have to modify inplace, not return another pointer
+            flux[:]=[internal_flux[j] if j in i and j!=i[-1] else prev for j,prev in enumerate(flux)]
+            #logging.info(f"{flux[:20]}")
+
+class ThickTargetModel(XspecModel):
+    def __init__(self):
+        self.ParInfo=(
+        "a0  1e-35  100.0  1.0  10.0  1e6  1e7  1.0",
+        "p  \"\"  4.0  1.1  1.5  15.0  20.0  0.1",
+           "eebrk  keV  150.0  1.0  5.0  100.  1e5  0.5" ,
+         "q  \"\"  6.0  0.0  1.5  15.0  20.0  0.1",
+         "eelow  keV  20.0  0.0  1.0  100.  1e3  1.0" ,
+         "eehigh  keV  3200.0  1.0  10.0  1e6  1e7  1.0"
+        ) #default parameters from OSPEX
+        self.model=self.thick2
+        self.description=f"Thick-target bremsstrahlung '{self.model.__name__}'"
+
+    @staticmethod
+    def thick2(photon_energies, params, flux):
```

Comment:
I think we should avoid inheriting ambiguous function names from IDL.  We should used self explanatory names like `bremsstrahlung_thick_target` and then in the docstring we can say that it's equivalent to `thick2`.